src/TypeScriptToKotlinWalker.kt

```diff
@@ -51,6 +51,7 @@ abstract class TypeScriptToKotlinBase : Visitor {
 
     val declarations = arrayListOf<Member>()
     val typesByTypeAlias = mutableMapOf<String,TS.TypeNode>()
+    val typeParameterDeclarations = mutableListOf<TS.TypeParameterDeclaration>()
```

Comment:
An object literal parameter can refer to type params for the function.  When that is the case, the T$ interface needs to declare those same type parameters.